java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -224,6 +221,23 @@ public HttpEventResponse handle(HttpEventRequest request) {
         }
     }
 
+    /**
+     * encapsulates dynamically configuring Sanitizer based on request (to support some aspects of
+     * its behavior being controlled via HTTP headers)
+     *
+     * @param request
+     */
+    void dynamicallyConfigureSanitizer(HttpEventRequest request) {
```

Comment:
I'd make this return the sanitizer so it is explicit there is a change, rather than recreate internally, it is hidden side-effect.
I'd do something like this:

```
Optional<Sanitizer> perRequestSanitizer = requestBasedSanitizer(request);
if (perRequestSanitizer.isPresent()) {
  sanitizer = perRequestSanitizer.get();
}
```